src/osx/carbon/cursor.cpp

```diff
@@ -278,7 +278,28 @@ wxCursor::wxCursor(const wxString& cursor_file, wxBitmapType flags, int hotSpotX
     if ( flags == wxBITMAP_TYPE_MACCURSOR_RESOURCE )
     {
 #if wxOSX_USE_COCOA
-        wxFAIL_MSG( wxT("Not implemented") );
+        wxString fileName;
+        wxImage image;
+        fileName = wxStandardPaths::Get().GetResourcesDir() + "/" + cursor_file + ".png";
+        image.LoadFile( fileName, wxBITMAP_TYPE_PNG );
+        if( image.IsOk() )
+        {
+            image.SetOption( wxIMAGE_OPTION_CUR_HOTSPOT_X, hotSpotX ) ;
+            image.SetOption( wxIMAGE_OPTION_CUR_HOTSPOT_Y, hotSpotY ) ;
+        }
+        else
+        {
+            fileName = wxStandardPaths::Get().GetResourcesDir() + "/" + cursor_file + ".cur";
+            image.LoadFile( fileName, wxBITMAP_TYPE_CUR );
+        }
+        if( image.IsOk() )
+        {
+            m_refData->DecRef() ;
+            m_refData = NULL ;
+            InitFromImage( image );
+        }
+        else
+            wxFAIL_MSG( "No PNG cursor image found in Resources" );
```

Comment:
This probably should be `wxLogError()`. The error message is also misleading because it tries both PNG and CUR.